src/Argu/Attributes.fs

```diff
@@ -128,6 +129,30 @@ type EqualsAssignmentAttribute () =
 [<AttributeUsage(AttributeTargets.Property, AllowMultiple = false)>]
 type ColonAssignmentAttribute () = 
     inherit CustomAssignmentAttribute(":")
+    
+/// Use an custom separator for parameter assignment.
+/// e.g. '--param<separator>arg' or '--param key<separator>value'.
+/// Parameters can also be assigned using space as separator e.g. '--param arg'
+/// Requires that the argument should have parameters of arity 1 or 2 only.
```

Comment:
What is the expected behaviour when arity is 2? Presumably both `--foo key=value` and `--foo key value` would be accepted? There should probably be tests for that.